libr/anal/p/anal_loongarch_gnu.c

```diff
@@ -0,0 +1,1338 @@
+			break;
+		case LA_INS_SLLI_D:
+			r_strbuf_appendf(&op->esil,"%d,%s,<<,%s,=", I_I6(opcode), LA_RJ(), LA_RD());
+			break;
+		case LA_INS_SRLI_D:
+			r_strbuf_appendf(&op->esil,"%d,%s,>>,%s,=", I_I6(opcode), LA_RJ(), LA_RD());
+			break;
+		case LA_INS_SRAI_D:
+			r_strbuf_appendf(&op->esil,"%d,%s,>>>>,%s,=", I_I6(opcode), LA_RJ(), LA_RD());
+			break;
+		case LA_INS_ROTRI_D:
+			r_strbuf_appendf(&op->esil,"%d,%s,>>>,%s,=", I_I6(opcode), LA_RJ(), LA_RD());
+			break;
+		case LA_INS_MOVE:
+			r_strbuf_appendf(&op->esil,"%s,%s,=",LA_RJ(), LA_RD());
+			break;
+		case LA_INS_AND:
+			r_strbuf_appendf(&op->esil,"%s,%s,&,%s,=", LA_RJ(), LA_RK(),LA_RD());
+			break;
+		case LA_INS_OR:
+			r_strbuf_appendf(&op->esil,"%s,%s,|,%s,=", LA_RJ(), LA_RK(),LA_RD());
+			break;
+		case LA_INS_XOR:
+			r_strbuf_appendf(&op->esil,"%s,%s,^,%s,=", LA_RJ(), LA_RK(),LA_RD());
+			break;
+		case LA_INS_NOR:
+			r_strbuf_appendf(&op->esil,"%s,%s,|,0xffffffff,^,%s,=", LA_RJ(), LA_RK(),LA_RD());
+			break;
+		case LA_INS_ANDN:
+			r_strbuf_appendf(&op->esil,"%s,^,0xffffffff,%s,&,%s,=", LA_RK(), LA_RJ(),LA_RD());
+			break;
+		case LA_INS_ORN:
+			r_strbuf_appendf(&op->esil,"%s,^,0xffffffff,%s,|,%s,=", LA_RK(), LA_RJ(),LA_RD());
+			break;
+		case LA_INS_ANDI:
+			r_strbuf_appendf(&op->esil,"%d,%s,&,%s,=", I_I12(opcode), LA_RJ(),LA_RD());
+			break;
+		case LA_INS_ORI:
+			r_strbuf_appendf(&op->esil,"%d,%s,|,%s,=", I_I12(opcode), LA_RJ(), LA_RD());
+			break;
+		case LA_INS_XORI:
+			r_strbuf_appendf(&op->esil,"%d,%s,^,%s,=", I_I12(opcode), LA_RJ(), LA_RD());
+			break;
+		case LA_INS_B:
+			r_strbuf_appendf(&op->esil,"0x%"LA_PFM",pc,+,pc,=", I26s2_SX(opcode));
+			break;
+		case LA_INS_BL:
+			r_strbuf_appendf(&op->esil,"4,pc,+,ra,=,0x%"LA_PFM",pc,+,pc,=", I26s2_SX(opcode));
+			break;
+		case LA_INS_JIRL:
+			r_strbuf_appendf(&op->esil,"4,pc,+,%s,=,0x%"LA_PFM",pc,+,pc,=",LA_RD(), I16s2_SX(opcode));
+			break;
+		case LA_INS_BEQ:
+			r_strbuf_appendf(&op->esil,"%s,%s,==,$z,?{,0x%"LA_PFM",pc,+,pc,=,}",LA_RJ(), LA_RD(), I16s2_SX(opcode));
+			break;
+		case LA_INS_BEQZ:
+			r_strbuf_appendf(&op->esil,"%s,0,==,$z,?{,0x%"LA_PFM",pc,+,pc,=,}",LA_RJ(),  I21s2_SX(opcode));
+			break;
+		case LA_INS_BNEZ:
+			r_strbuf_appendf(&op->esil,"%s,0,==,$z,!,?{,0x%"LA_PFM",pc,+,pc,=,}",LA_RJ(),  I21s2_SX(opcode));
+			break;
+		case LA_INS_BNE:
+			r_strbuf_appendf(&op->esil,"%s,%s,==,$z,!,?{,0x%"LA_PFM",pc,+,pc,=,}",LA_RJ(), LA_RD(), I16s2_SX(opcode));
+			break;
+			/* FIXME U means unsigned comparison*/
+		case LA_INS_BLTU:
+		case LA_INS_BLT:
+			r_strbuf_appendf(&op->esil,"%s,%s,<,?{,0x%"LA_PFM",pc,+,pc,=,}",LA_RD(), LA_RJ(), I16s2_SX(opcode));
+			break;
+		case LA_INS_BGEU:
+		case LA_INS_BGE:
+			r_strbuf_appendf(&op->esil,"%s,%s,>,?{,0x%"LA_PFM",pc,+,pc,=,}",LA_RD(), LA_RJ(), I16s2_SX(opcode));
+			break;
+
+		default:
+			break;
+	}
+	return 0;
+}
+static int 
+loongarch_op(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *b, int len, RAnalOpMask mask) {
+	
```

Comment:
handle the DISASM mask and remove the asm plugin :) that will save me some refactoring time after mergnig

Reply:
okay.But for gnu disassembly it seems to need a little bit more interface to the asm plugin.

Reply:
cs asm is easier to do